src/main/java/com/salesforce/mirus/KafkaMonitor.java

```diff
@@ -186,19 +192,19 @@ public void run() {
           return;
         }
         consecutiveRetriableErrors = 0;
+      } catch (WakeupException | InterruptedException e) {
```

Comment:
Consider [this guidance from SonarQube](https://next.sonarqube.com/sonarqube/coding_rules?open=squid%3AS2142&rule_key=squid%3AS2142) about the proper handling of `InterruptedException`s, in particular calling `Thread.currentThread().interrupt()` after catching.

Reply:
Per our discussion, maybe it's best not to interrupt the thread again, since we're looping.